Exercise_2.java

```diff
@@ -1,52 +1,116 @@
-public class StackAsLinkedList { 
-  
-    StackNode root; 
-  
-    static class StackNode { 
-        int data; 
-        StackNode next; 
-  
-        StackNode(int data) 
-        { 
-            //Constructor here 
-        } 
-    } 
-    
-	
-    public boolean isEmpty() 
-    { 
-        //Write your code here for the condition if stack is empty. 
-    } 
-  
-    public void push(int data) 
-    { 
-        //Write code to push data to the stack. 
-    } 
-  
-    public int pop() 
-    { 	
-	//If Stack Empty Return 0 and print "Stack Underflow"
+/* Approach - var length to check the overflow and underflow of stack.
+traversing till the node.next is null to perform the push pop peek operations
+
+Time Complxity - O(n)
+Space - O(n)
+
+Optmization - Using array or Doubly linkedlist will reduce time cost to O(1). Space will still be O(n)
+*/
+
+
+
+
+public class StackAsLinkedList {
+
+    StackNode root;
+    int length;
+
+    static class StackNode {
+        int data;
+        StackNode next;
+
+        StackNode(int dataa)
+        {
+            data = dataa;
+            next = null;
+
+        }
+    }
+    StackAsLinkedList()
+    {
+        root = new StackNode(-1);
+        length = 0;
+    }
+
+    public boolean isEmpty()
+    {
+        //Write your code here for the condition if stack is empty.
+        if(length ==0) return true;
+        return false;
+    }
+
+    public void push(int data)

```

Comment:
See, You have to implement stack using LinkedList,Time for intert/delete should be O(1) according to stack operations. So you should insert/delete element from head of the list. So then it will be Called as stack with time O(1).